### PR TITLE
should restore cookie correctly when using useRole with the preserveUrl option (closes #2282)

### LINKED ISSUE #2282: Cookies are not restored correctly when User Roles with the preserveUrl option are used
### Are you requesting a feature or reporting a bug?

bug

### What is the current behavior?

Cookies are not restored correctly when User Roles with the preserveUrl option are used

### What is the expected behavior?

Cookies should be restored correctly when User Roles with the preserveUrl option are used

### PR DESCRIPTION
**Pull request by AlexKamaev**

### CODE REVIEW

File: src/test-run/commands/actions.js [hunk 1]
```diff
@@ -303,17 +303,17 @@ export class NavigateToCommand extends Assignable {
...
+        this.stateSnapshot = null;
 
         this._assignFrom(obj, true);
     }
 
     _getAssignableProperties () {
         return [
             { name: 'url', type: urlArgument, required: true },
-            { name: 'storages', type: nullableStringArgument }
+            { name: 'stateSnapshot', type: nullableStringArgument }
```
Comment (helen-dikareva) [inline src/test-run/commands/actions.js]: `stateSnapshot` isn't 'string or null' anymore, is it?

File: test/functional/fixtures/regression/gh-2282/pages/authorized.html [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>Title</title>
+    <script type="text/javascript">
+        window.onload = function () {
+            if (!document.cookie || document.cookie === 'auth=false')
+                location.href = './login.html';
+            else {
+                var result = document.getElementById('result');
```
Comment (helen-dikareva) [inline test/functional/fixtures/regression/gh-2282/pages/authorized.html]: please add new line after new var

File: test/functional/fixtures/regression/gh-2282/testcafe-fixtures/index.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+const role = Role('http://localhost:3000/fixtures/regression/gh-2282/pages/login.html', async t => {
+    await t.click(Selector('#login'));
+}, { preserveUrl: true });
+
+
+fixture `GH-2282 - Cookies should be restored correctly when User Roles with the preserveUrl option are used`;
+
+test('Login and save cookies in the role', async t => {
+    await t
+        .useRole(role)
```
Comment (helen-dikareva) [inline test/functional/fixtures/regression/gh-2282/testcafe-fixtures/index.js]: Can we move `useRole` to `fixture.beforeEach`?

File: src/test-run/commands/actions.js (hunk 1 above)
Comment (AlexKamaev) [reply src/test-run/commands/actions.js]: still is

File: test/functional/fixtures/regression/gh-2282/testcafe-fixtures/index.js (hunk 3 above)
Comment (AlexKamaev) [reply test/functional/fixtures/regression/gh-2282/testcafe-fixtures/index.js]: yes

File: src/test-run/commands/actions.js (hunk 1 above)
Comment (helen-dikareva) [reply src/test-run/commands/actions.js]: [yep](https://github.com/DevExpress/testcafe/pull/2285/files#diff-ef5c498529d63a584dbbfbbeeada71e1R110), I see now

**Pull request merged**